cmd/paramgen/internal/paramgen.go

```diff
@@ -0,0 +1,628 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"os/exec"
+	"reflect"
+	"strconv"
+	"strings"
+	"unicode"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+const (
+	TagParamName         = "json"
+	TagParamDefault      = "default"
+	TagParamValidate     = "validate"
+	tagRequired          = "required"
+	tagSeparator         = ","
+	validateTagSeparator = "="
+	tagLessThan1         = "lt"
+	tagLessThan2         = "less-than"
+	tagGreaterThan1      = "gt"
+	tagGreaterThan2      = "greater-than"
+	tagInclusion         = "inclusion"
+	tagExclusion         = "exclusion"
+	pipeSeparator        = "|"
+)
+
+// ParseParameters parses the struct into a map of parameter, requires the folder path that has the struct, and the
+// struct name
+func ParseParameters(path string, name string) (map[string]sdk.Parameter, error) {
```

Comment:
Btw, seems like this is the only thing we need to expose, we could make everything else private (constants, `Module`, `ModuleError`).